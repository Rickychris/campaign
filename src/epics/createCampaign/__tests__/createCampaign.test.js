import React from "react";
import ReactDom from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import CreateCampaign from "../createCampaign";

describe("CreateCampaign  component tests", () => {
  beforeEach(cleanup);
  const CreateCampaignTest = (
    <BrowserRouter>
      <CreateCampaign />
    </BrowserRouter>
  );
  it("renders the CreateCampaign component", () => {
    const div = document.createElement("div");
    ReactDom.render(CreateCampaignTest, div);
  });

  it("renders the CreateCampaign with text correctly", () => {
    const { getByTestId } = render(CreateCampaignTest);
    expect(getByTestId("create-campaign")).toHaveTextContent(
      "Your Campaign Name"
    );
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(CreateCampaignTest);
    expect(getByTestId("create-campaign")).toMatchSnapshot();
  });
});
