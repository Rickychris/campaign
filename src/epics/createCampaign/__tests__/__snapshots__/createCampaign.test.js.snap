// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCampaign  component tests matches snapshot 1`] = `
<div
  class="root"
  data-testid="create-campaign"
>
  <form
    id="campaign-form-id"
  >
    <h3
      class="form-sub-header"
    >
      Your Campaign Name
    </h3>
    <input
      placeholder="Enter the Campaign Name"
      required=""
      type="text"
      value=""
    />
    <div
      class="root"
      data-testid="spacer"
      style="height: 40px;"
    />
    <h3
      class="form-sub-header"
    >
      Email Target List
    </h3>
    <div
      class=" react-multi-email  empty"
    >
      <span
        data-placeholder="true"
      >
        Enter your emails
      </span>
      <input
        type="text"
        value=""
      />
    </div>
    <div
      class="root"
      data-testid="spacer"
      style="height: 80px;"
    />
    <h3
      class="form-sub-header"
    >
      From
    </h3>
    <input
      placeholder="Enter your Name"
      required=""
      type="text"
      value=""
    />
    <input
      placeholder="Enter your Email"
      required=""
      type="email"
      value=""
    />
    <div
      class="root"
      data-testid="spacer"
      style="height: 40px;"
    />
    <h3
      class="form-sub-header"
    >
      Email Details
    </h3>
    <input
      placeholder="Add the Subject for your Email"
      required=""
      type="text"
      value=""
    />
    <textarea
      placeholder="Add the Email Body"
      required=""
      type="text"
    />
    <button
      class="root"
      data-testid="button"
      text="Create Campaign"
      type="submit"
    >
      Create Campaign
    </button>
  </form>
</div>
`;
